app/api/events.py

```diff
@@ -140,8 +206,44 @@ def before_get_object(self, view_kwargs):
                 else:
                     view_kwargs['id'] = None
 
-    decorators = (jwt_required, )
+    def after_get(self, result):
+        result = changeToTimezone(result)
+
+    def after_patch(self, result):
+        result = changeToTimezone(result)
+
+    def after_post(self, result):
+        result = changeToTimezone(result)
+
+    decorators = (jwt_required,)
     schema = EventSchema
     data_layer = {'session': db.session,
                   'model': Event,
-                  'methods': {'before_get_object': before_get_object}}
+                  'methods': {'before_get_object': before_get_object
+                              }
+                  }
+
+
+class EventRelationship(ResourceRelationship):
+    decorators = (jwt_required,)
+
+    schema = EventSchema
+    data_layer = {'session': db.session,
+                  'model': Event}
+
+
+def changeToTimezone(result):
+    tz = timezone(result['data']['attributes']['timezone'])
+
+    start_date_datetime = tz.localize(parser.parse(result['data']['attributes']['starts_at']).replace(tzinfo=None))
+    result['data']['attributes']['starts_at'] = start_date_datetime.isoformat()
+
+    end_date_datetime = tz.localize(parser.parse(result['data']['attributes']['ends_at']).replace(tzinfo=None))
+    result['data']['attributes']['ends_at'] = end_date_datetime.isoformat()
+
+    if result['data']['attributes']['schedule_published_on']:
+        schedule_pub_datetime = tz.localize(parser.parse(result['data']['attributes']['schedule_published_on']).replace(tzinfo=None))
+        result['data']['attributes']['schedule_published_on'] = schedule_pub_datetime.isoformat()
+
+
+    return result
```

Comment:
@shubham-padia since it was discussed that one should store DateTime with timezone, should it be done in this issue itself for the events API of will everything be solved later on in #3739? cause if so, we do not have to go all the way to create this 'changeToTimezone' function, is it?